src/utils/validateNric.tsx

```diff
@@ -35,19 +36,24 @@ const validate = (nricInput: string): boolean => {
   }
 
   // if the nric type is T or G, add 4 to the total
+  // else if the nric type is M, add 3 to the total
   if (["T", "G"].indexOf(nricType) >= 0) {
     total += 4;
+  } else if (["M"].indexOf(nricType) >= 0) {
+    total += 3;
   }
 
   // check last letter of nric for local
   const letterIndex = total % 11;
   const nricLetter = nricArr[3];
   if (["S", "T"].indexOf(nricType) >= 0) {
     return nricLetterST[letterIndex] === nricLetter;
+  } else if (["T", "G"].indexOf(nricType) >= 0) {
+    // check last letter of nric for foreigners
+    return nricLetterFG[letterIndex] === nricLetter;
```

Comment:
Is this logic new?

Reply:
nope, should we modernize the whole logic?

Reply:
Not sure if `["T", "G"].indexOf(nricType)` should be `["F", "G"].indexOf(nricType)` too!

Reply:
Oh, noticed you just moved this logic up, my bad. Make sure to check my previous comment though!

Reply:
nope, it's T and G!
> 2) If the first letter of the NRIC starts with T or G, add 4 to the total.